lib/collection/src/operations/types.rs

```diff
@@ -183,6 +184,37 @@ impl TryFrom<Record> for PointStructPersisted {
     }
 }
 
+// Version of the collection config we can present to the user
+/// Information about the collection configuration
+#[derive(Debug, Serialize, JsonSchema)]
+pub struct CollectionConfig {
+    #[validate(nested)]
+    pub params: CollectionParams,
+    #[validate(nested)]
+    pub hnsw_config: HnswConfig,
+    #[validate(nested)]
+    pub optimizer_config: OptimizersConfig,
+    #[validate(nested)]
+    pub wal_config: Option<WalConfig>,
+    #[serde(default)]
+    pub quantization_config: Option<QuantizationConfig>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub strict_mode_config: Option<StrictModeConfig>,
+}
+
+impl From<CollectionConfigInternal> for CollectionConfig {
+    fn from(config: CollectionConfigInternal) -> Self {
+        CollectionConfig {
+            params: config.params,
+            hnsw_config: config.hnsw_config,
+            optimizer_config: config.optimizer_config,
+            wal_config: Some(config.wal_config),
+            quantization_config: config.quantization_config,
+            strict_mode_config: config.strict_mode_config,
+        }
```

Comment:
Shall we do exhaustive destructing here of `CollectionConfigInternal`?

That way we're sure we don't accidentally forget fields exposing fields if we change the internal structure.

```suggestion
        let CollectionConfigInternal {
            params,
            hnsw_config,
            optimizer_config,
            wal_config,
            quantization_config,
            strict_mode_config,
            // Internal UUID to identify unique collections in consensus snapshots
            uuid: _,
        } = config;

        CollectionConfig {
            params,
            hnsw_config,
            optimizer_config,
            wal_config: Some(wal_config),
            quantization_config,
            strict_mode_config,
        }
```